packages/atomic/src/components/commerce/product-template-components/atomic-product-image/e2e/atomic-product-image.e2e.ts

```diff
@@ -1,12 +1,353 @@
+      expect(accessibilityResults.violations.length).toEqual(0);
+    });
+
+    test('should use the default alt text for all images', async ({
+      productImage,
+    }) => {
+      expect(await productImage.noCarouselImage.getAttribute('alt')).toEqual(
+        'Image 1 out of 1 for Nublu Water Bottle'
+      );
+      expect(await productImage.carouselImage.getAttribute('alt')).toEqual(
+        'Image 1 out of 2 for Blue Lagoon Mat'
+      );
+      await productImage.nextButton.click();
+      await expect
+        .poll(async () => {
+          return await productImage.carouselImage.getAttribute('alt');
+        })
+        .toContain('Image 2 out of 2 for Blue Lagoon Mat');
+    });
+  });
+});
+
+test.describe('as a carousel', async () => {
+  const URL =
+    'http://localhost:4400/iframe.html?id=atomic-product-image--default&viewMode=story#sortCriteria=relevance';
+  const FIRST_IMAGE =
+    'https://images.barca.group/Sports/mj/Trampolines%20%26%20Floats/Huge%20inflatable%20mats/3_Blue/df1a99488425_bottom_right.webp';
+  const SECOND_IMAGE =
+    'https://images.barca.group/Sports/mj/Trampolines%20%26%20Floats/Huge%20inflatable%20mats/3_Blue/df1a99488425_bottom_left.webp';
+
+  test.beforeEach(async ({productImage}) => {
+    await productImage.load();
+    await productImage.carouselImage.waitFor();
+  });
+
+  test('should be accessible', async ({makeAxeBuilder}) => {
+    const accessibilityResults = await makeAxeBuilder().analyze();
+    expect(accessibilityResults.violations.length).toEqual(0);
+  });
+
+  test('should render the first image by default', async ({productImage}) => {
+    await expect(productImage.carouselImage).toBeVisible();
+    const src = await productImage.carouselImage.getAttribute('src');
+    expect(src).toContain(FIRST_IMAGE);
+  });
+
+  test.describe('when clicking the next button', () => {
+    test.beforeEach(async ({productImage}) => {
+      await productImage.nextButton.click();
+    });
+
+    test('should navigate to the next image', async ({productImage}) => {
+      await expect
+        .poll(async () => {
+          const src = await productImage.carouselImage.getAttribute('src');
+          return src;
+        })
+        .toContain(SECOND_IMAGE);
+    });
+
+    test('should not open the product', async ({page}) => {
+      expect(page.url()).toEqual(URL);
+    });
+  });
+
+  test.describe('when clicking the previous button', () => {
+    test.beforeEach(async ({productImage}) => {
+      await productImage.previousButton.click();
+    });
+
+    test('should navigate to the previous image', async ({productImage}) => {
+      await expect
+        .poll(async () => {
+          const src = await productImage.carouselImage.getAttribute('src');
+          return src;
+        })
+        .toContain(SECOND_IMAGE);
+    });
+
+    test('should not open the product', async ({page}) => {
+      expect(page.url()).toEqual(URL);
+    });
+  });
```

Comment:
A few additional tests we should add IMO:

- when clicking the next button, if the current image is the last one should render the first image
- when clicking the back button, if the current image is the first onne, should render the last image
- when clicking an indicator dot, should render the corresponding image

Reply:
thanks forgot about the indicator dot